src/components/AdminSettings/AdminModal.jsx

```diff
@@ -0,0 +1,124 @@
+import { useEffect, useState } from 'react';
+import { useBackend } from '../../contexts/BackendContext';
+import PropTypes from 'prop-types';
+import {
+  Alert,
+  AlertTitle,
+  AlertDescription,
+  Button,
+  IconButton,
+  Input,
+  Flex,
+  Text,
+  Modal,
+  ModalContent,
+  ModalHeader,
+  ModalBody,
+  ModalOverlay,
+  ModalFooter,
+} from '@chakra-ui/react';
+import { CloseIcon } from '@chakra-ui/icons';
+
+// TODO: Refactor states such that its a nested state
+const AdminModal = ({ isOpen, onClose, data, loadInfo }) => {
+  const { backend } = useBackend();
+  const [alertVisible, setAlertVisible] = useState(false);
+  const [nameData, setNameData] = useState('');
+  const [emailData, setEmailData] = useState('');
+  const [lastUpdatedData, setLastUpdatedData] = useState(new Date().toISOString());
+
+  useEffect(() => {
+    const getData = async () => {
+      try {
+        if (isOpen && data) {
+          setNameData(data['name']);
+          setEmailData(data['email']);
+          setLastUpdatedData(data['last_updated']);
+        }
+      } catch (error) {
+        console.error('Error fetching data:', error);
+      }
+    };
+    getData();
+  }, [isOpen]);
```

Comment:
⚠️ **[eslint]** <[react-hooks/exhaustive-deps](https://github.com/facebook/react/issues/14920)> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>React Hook useEffect has a missing dependency: 'data'. Either include it or remove the dependency array. If 'setNameData' needs the current value of 'data', you can also switch to useReducer instead of useState and read 'data' in the reducer.